### PR TITLE
fix(ZNTA-628): fall back on locale's plural forms

### PR DESCRIPTION
**Pull request by seanf**
Fall back on locale's plural forms when plurals header is empty.
Also simplify the code and improve log messages.

### CODE REVIEW

**Comment by carlosmunoz** [comment]
:+1: Reviewed

**Comment by definite** [comment]
@zanata-jenkins retest this please

**Comment by definite** [comment]
Test Cases:
:+1:  Header with "Plural-Forms="
:x: Header with "Plural-Forms:"
     
Expected: 
  "Plural-Froms: <locale's plural froms>"

**Comment by seanf** [comment]
@carlosmunoz I added another commit.

File: zanata-war/src/main/java/org/zanata/rest/service/ResourceUtils.java [hunk 1]
```diff
@@ -768,7 +773,7 @@ private void populateHeaderEntries(final List<HeaderEntry> headerEntries,
         }
 
         headerEntry = containedHeaders.get(PLURAL_FORMS_HDR);
-        if (headerEntry == null) {
+        if (headerEntry == null || headerEntry.getValue().isEmpty()) {
```
Comment (carlosmunoz) [inline zanata-war/src/main/java/org/zanata/rest/service/ResourceUtils.java]: I think a call to `trim()` here might be good too... unless the entry is being trimmed elsewhere already.

**Comment by carlosmunoz** [comment]
Added a comment to the new commit. Review complete.

**Comment by seanf** [comment]
@carlosmunoz I've changed the check to account for blank strings too.

**Comment by carlosmunoz** [comment]
Thanks Sean. Looks good to me.

**Comment by definite** [comment]
Test Cases:
:+1:  Header with "Plural-Forms="
:+1: Header with "Plural-Forms:"

VERIFIED

**Pull request merged**